### PR TITLE
[8.x] Make the foreign id column type changeable

### PR DESCRIPTION
**Pull request by mtawil**

### CODE REVIEW

File: src/Illuminate/Database/Schema/Blueprint.php [hunk 1]
```diff
@@ -836,15 +836,16 @@ public function unsignedBigInteger($column, $autoIncrement = false)
...
     /**
-     * Create a new unsigned big integer (8-byte) column on the table.
+     * Create a new foreign id column on the table with a specific type.
      *
      * @param  string  $column
+     * @param  string  $type
      * @return \Illuminate\Database\Schema\ForeignIdColumnDefinition
      */
-    public function foreignId($column)
+    public function foreignId($column, $type = 'bigInteger')
```
Comment (driesvints) [inline src/Illuminate/Database/Schema/Blueprint.php:845]: This is a breaking change.

Comment (mtawil) [reply src/Illuminate/Database/Schema/Blueprint.php:845]: How so? I just set the default value.

Comment (driesvints) [reply src/Illuminate/Database/Schema/Blueprint.php:845]: Anyone who is overwriting this method will see their implementation break.

Comment (mtawil) [reply src/Illuminate/Database/Schema/Blueprint.php:845]: Oh didn't notice! Thank you, will making a new PR to master

**Pull request closed without merging**